### ISSUE TITLE
Keep getting getsockopt failures after leaving system for a while

### ISSUE THREAD

**Issue by prashvin** [contiv/netplugin#291]
Any scripts to reset this state ? Keep getting stuck at this issue with the vagrant box.

vagrant@netplugin-node1:~$ netctl policy create prod_web
INFO[0000] Creating policy default:prod_web             
ERRO[0000] Post http://localhost:9999/api/policys/default:prod_web/: dial tcp [::1]:9999: getsockopt: connection

**Comment by prashvin** [comment]
This is the nohup.out error message: 
updating key /contiv.io/service/netplugin/192.168.2.10:9002, Err: 100: Key not found (/contiv.io/service/netplugin/192.168.2.10:9002

**Comment by vvb** [comment]
@prashvin Can you upload the ~/nohup.out to a gist and share the link?

**Comment by vvb** [comment]
and also the output of the following commands,
```
docker version
netplugin version
sudo yum install -y lsof
sudo lsof -p $(pidof netplugin) | wc -l
```

**Comment by jainvipin** [comment]
@prashvin - for vagrant environment, `make host-restart` in vagrant environment does the cleanup. It utilizes `cleanup.py` or `startPlugin.py` in `scripts/python` directory. 
It would be good to know the steps that results into this issue/state, sharing the ~/nohup.out would help determine that.

@vvb, I think you meant `netctl version` or `netplugin --verison` and not `netplugin version`

**Comment by vvb** [comment]
@jainvipin thanks, edited my comment to change it to `netctl version`

**Comment by prashvin** [comment]
Gist file here. Doing a make demo again resets and makes the netctcl happy again. Make host-restart had some issues (put the issue in gist file)
https://gist.github.com/prashvin/28d8a01b73ec29a8f47c

**Comment by jainvipin** [comment]
@prashvin - `make host-restart` only works if you bring the system with GOPATH set i.e. if it is a development environment as mentioned [here](https://github.com/contiv/netplugin/blob/master/docs/DevEnv.md) 

Looking at the logs.... meanwhile can you share what you did to get into this situation?

**Comment by jainvipin** [comment]
ping @shaleman - seems like netmaster ha related changes (recent ones)

**Comment by shaleman** [comment]
Thanks @prashvin 

Looks like one of the vagrant VMs was SIGSTOPed for a while and some of the etcd entries TTL expired behind netmaster/netplugin's back and we failed to recover from that state. This is a bug. We'll fix it soon. meanwhile, please use `make demo` or `make host-restart` workaround.

PS: `make host-restart` works only inside the vagrant VM. Please ssh to vagrant VM, go to `/opt/gopath/src/github.com/contiv/netplugin/` and then run `make host-restart`. More info in the doc linked by @jainvipin 

Relavent errors:
```
time="Feb 21 16:20:04.570043060" level=error msg="Error updating key /contiv.io/service/netmaster/192.168.2.10:9999, Err: 100: Key not found (/contiv.io/service/netmaster/192.168.2.10:9999) [688]"
time="Feb 21 16:20:05.432149327" level=error msg="Error updating TTl. Err: 100: Key not found (/contiv.io/lock/netmaster/leader) [688]"
time="Feb 21 16:20:14.177019854" level=error msg="Error updating key /contiv.io/service/netplugin/192.168.2.10:9002, Err: 100: Key not found (/contiv.io/service/netplugin/192.168.2.10:9002) [688]"
time="Feb 21 16:20:34.183259904" level=error msg="Error updating key /contiv.io/service/netplugin/192.168.2.10:9002, Err: 100: Key not found (/contiv.io/service/netplugin/192.168.2.10:9002) [690]"
```

**Comment by shaleman** [comment]
This is fixed by #325 
Please reopen if you still see the issue
